config.py

```diff
@@ -0,0 +1,629 @@
+
+        # Skip if very first chunk
+        if self.job_ids['previous']:
+            # Wait for previous chunk to be done
+            self.wait_for_previous()
+
+            # Get and print previous chunk Slurm summary
+            self.get_previous_slurm_summary()
+            self.print_previous_slurm_summary()
+
+            # Check for success of all previous jobs
+            self.check_previous_chunk_success()
+
+        # Cycle info
+        self.job_ids['previous'] = self.job_ids['current']
+        self.previous_chunk_id = self.chunk_id
+
+        # Monitor last chunk
+        if self.enddate_sim >= self.enddate:
+            self.wait_for_previous()
+            self.get_previous_slurm_summary()
+            self.print_previous_slurm_summary()
+            self.check_previous_chunk_success()
+
+    @staticmethod
+    def get_job_info(job_id,
+                     slurm_keys=['JobName', 'Elapsed', 'ExitCode'],
+                     parse=True):
+        """Retrieve slurm job info as given by sacct
+
+        if parse is True, return the raw string from sacct else parse info into a dict.
+        All possible keys are given by `sacct --helpformat`"""
+
+        # Get info from sacct
+        cmd = ["sacct", f"--format={', '.join(slurm_keys)}", "-j", str(job_id)]
+
+        if parse:
+            cmd.append("--parsable")
+
+        info_str = run(cmd, capture_output=True, check=True).stdout
+
+        if parse:
+            # Parse in a dictionnary before returning
+            # The inner most process should be the relevant one, hence the 1 index
+            slurm_info = info_str.split(b'\n')[1].split(b'|')
+            return {k: v.decode() for k, v in zip(slurm_keys, slurm_info)}
+        else:
+            return info_str.decode()
+
+    def get_previous_slurm_summary(self,
+                                   info_keys=[
+                                       'JobName', 'JobID', 'Partition',
+                                       'NNodes', 'State', 'Start', 'End',
+                                       'Elapsed'
+                                   ]):
+        """get slurm info summary or all jobs of previous chunk"""
+
+        # Store requested keys in object
+        self.info_keys = info_keys
+
+        # Get job info for all jobs
+        self.slurm_info = {}
+        for job_name in self.jobs:
+            for job_id in self.job_ids['previous'][job_name]:
+                self.slurm_info[job_name] = []
+                self.slurm_info[job_name].append(
+                    self.get_job_info(job_id, slurm_keys=info_keys,
+                                      parse=True))
+
+    def print_previous_slurm_summary(self):
+        # Width of printed slurm piece of information
+        info_width = {
+            'JobName': 13,
+            'JobID': 8,
+            'Partition': 9,
+            'NNodes': 3,
+            'State': 14,
+            'Start': 13,
+            'End': 13,
```

Comment:
'Start' and 'End' field width will be too small in case a job runs past midnight. This will happen as soon as the processing_chain gets used for production on climate simulations.

Reply:
What would be the correct/optimal width?

Reply:
I think I set it to the right width first and you changed it in a commit afterwards so git blaming yourself should work :-P

Reply:
GitHub blames [GitHub Action: Apply Pep8-formatting](https://github.com/C2SM/processing-chain/pull/54/commits/c9846beaa2155156addc2b67422a9cc562049efc) 😞

Reply:
To me it looks like we can get 6 additional spaces: 5 from `State` and one from `Elapsed`:

```
      JobName    JobID Partition  #N          State         Start           End   Elapsed
------------- -------- --------- --- -------------- ------------- ------------- ---------
 prepare_icon 51627355    normal   1      COMPLETED      13:36:28      13:37:25  00:00:57
prepare_art_g 51627356    normal   1      COMPLETED      13:38:31      13:38:45  00:00:14
         icon 51627357    normal   4      COMPLETED      13:38:50      13:39:09  00:00:19
```

Then, we can increase `Start` and `End` to 16 spaces.

Reply:
Also this can be done in a separate PR.